### PR TITLE
test: Migrated root smoke tests to `node:test`

### LINKED ISSUE #2756: Migrate root smoke tests to `node:test`
### Description
Migrate the following smoke tests to `node:test`:

 * client-s3.tap.js
 * lasp_connect.tap.js
 * proxy-api-connection-auth.tap.js
 * proxy-api-connection-noproxy.tap.js
 * s3-presigned-url.tap.js

### PR DESCRIPTION
**Pull request by bizob2828**
## Related Issues
Closes #2756

### CODE REVIEW

File: test/smoke/s3-presigned-url.test.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+'use strict'
+const test = require('node:test')
+const assert = require('node:assert')
+const helper = require('../lib/agent_helper')
+const https = require('https')
+
+test('@aws-sdk/s3-request-presigner functionality', (t, end) => {
+  const { version, name } = require('@aws-sdk/s3-request-presigner/package')
+  // eslint-disable-next-line no-console
+  console.log(`AWS package: ${name} version: ${version}`)
```
Comment (jsumners-nr) [inline test/smoke/s3-presigned-url.test.js:15]: Why are we logging this?

File: test/smoke/client-s3.test.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+const assert = require('node:assert')
+const helper = require('../lib/agent_helper')
+const {
+  DESTINATIONS: { TRANS_SEGMENT }
+} = require('../../lib/config/attribute-filter')
+
+test('@aws-sdk/client-s3 functionality', async (t) => {
+  const { version, name } = require('@aws-sdk/client-s3/package')
+  // eslint-disable-next-line no-console
+  console.log(`AWS package: ${name} version: ${version}`)
```
Comment (jsumners-nr) [inline test/smoke/client-s3.test.js:17]: Why are we logging this?

File: test/smoke/s3-presigned-url.test.js (hunk 1 above)
Comment (bizob2828) [reply test/smoke/s3-presigned-url.test.js:15]: we do caret match on the dep on install and in the past we had issues and it was hard to see which version of the dep caused the issue

File: test/smoke/client-s3.test.js (hunk 2 above)
Comment (bizob2828) [reply test/smoke/client-s3.test.js:17]: we do caret match on the dep on install and in the past we had issues and it was hard to see which version of the dep caused the issue

**Review by jsumners-nr — APPROVED** [review]

**Pull request merged**